xun/tests/test_graph.py

```diff
@@ -0,0 +1,120 @@
+from xun.functions.graph import CallNode
+from xun.functions.graph import CallNodeSubscript
+
+
+def test_unpack():
+    cn = CallNode('f')
+
+    assert CallNodeSubscript(cn, (0, )) == CallNodeSubscript(cn, (0, ))
+    assert not CallNodeSubscript(cn, (0, )) == CallNodeSubscript(cn, (1, ))
+
+    shape = (1, )
+    a = cn.unpack(shape)
+    required = (CallNodeSubscript(cn, (0, )), )
+    assert a == required
+
+    shape = (3, )
+    a = cn.unpack(shape)
+    required = (
+        CallNodeSubscript(cn, (0, )),
+        CallNodeSubscript(cn, (1, )),
+        CallNodeSubscript(cn, (2, )),
+    )
+    assert a == required
+
+    shape = (1, (2,))
+    a = cn.unpack(shape)
+    required = (
+        CallNodeSubscript(cn, (0, )),
+        (
+            CallNodeSubscript(cn, (1, 0)),
+            CallNodeSubscript(cn, (1, 1)),
+        ),
+    )
+    assert a == required
+
+    shape = (1, 1, (2,))
```

Comment:
Could we allow:
```suggestion
    shape = (2, (2,))
```
?